pkg/controllers/cronfederatedhpa/cronfederatedhpa_controller.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2023 The Karmada Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronfederatedhpa
+
+import (
+	"context"
+	_ "time/tzdata"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	autoscalingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/autoscaling/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/sharedcli/ratelimiterflag"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+const (
+	// ControllerName is the controller name that will be used when reporting events.
+	ControllerName = "cronfederatedhpa-controller"
+)
+
+// CronFHPAController is used to operate CronFederatedHPA.
+type CronFHPAController struct {
+	client.Client // used to operate Cron resources.
+	EventRecorder record.EventRecorder
+
+	RateLimiterOptions ratelimiterflag.Options
+	CronHandler        *CronHandler
+}
+
+// Reconcile performs a full reconciliation for the object referred to by the Request.
+// The Controller will requeue the Request to be processed again if an error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (c *CronFHPAController) Reconcile(ctx context.Context, req controllerruntime.Request) (controllerruntime.Result, error) {
+	klog.V(4).Infof("Reconciling CronFederatedHPA %s", req.NamespacedName.String())
+
+	cronFHPA := &autoscalingv1alpha1.CronFederatedHPA{}
+	if err := c.Client.Get(ctx, req.NamespacedName, cronFHPA); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(4).Infof("Begin to cleanup the cron jobs for CronFederatedHPA:%s", req.NamespacedName)
+			c.CronHandler.StopCronFHPAExecutor(req.NamespacedName.String())
+			return controllerruntime.Result{}, nil
+		}
+
+		klog.Error("Failed to get CronFederatedHPA(%s):", req.NamespacedName.String(), err)
+		return controllerruntime.Result{Requeue: true}, err
+	}
+
+	var err error
+	defer func() {
+		if err != nil {
+			c.EventRecorder.Event(cronFHPA, corev1.EventTypeWarning, "ProcessCronFederatedHPAFailed", err.Error())
+		}
+	}()
+
+	// If this CronFHPA is deleted, remove all the cron executors
```

Comment:
```suggestion
	// If this CronFederatedHPA is deleting, stop all related cron executors
```
1. We'd better not use the short name of `CronFederatedHPA `,
2. The object hasn't been `deleted` by now.